### PR TITLE
Fixed multiline input

### PR DESCRIPTION
**Pull request by dj95**
Fixed [Issue #107](https://github.com/LemonBoy/bar/issues/107).

Added wm class string, so it's easier to handle lemonbar in compton, xdotool, etc.

### CODE REVIEW

File: lemonbar.c [hunk 1]
```diff
@@ -1357,9 +1362,11 @@ main (int argc, char **argv)
                 else break;                         // ...bail out
             }
             if (pollin[0].revents & POLLIN) { // New input, process it
-                if (fgets(input, sizeof(input), stdin) == NULL)
-                    break; // EOF received
-
+                *input = 0;
+                while (fgets(input, sizeof input, stdin)) { }
+                if (!*input) {
```
Comment (LemonBoy) [inline lemonbar.c]: No need for this, this case is already handled by `parse`. Moreover this way you don't clear the bar when an empty string is read.

File: lemonbar.c [hunk 2]
```diff
@@ -1357,9 +1362,11 @@ main (int argc, char **argv)
                 else break;                         // ...bail out
             }
             if (pollin[0].revents & POLLIN) { // New input, process it
-                if (fgets(input, sizeof(input), stdin) == NULL)
-                    break; // EOF received
-
+                *input = 0;
+                while (fgets(input, sizeof input, stdin)) { }
```
Comment (LemonBoy) [inline lemonbar.c]: Please rewrite this as
```c
while (fgets(input, sizeof(input), stdin) != NULL)
    ; // Drain the buffer, the last line is actually used
```

File: lemonbar.c [hunk 3]
```diff
@@ -1357,9 +1362,11 @@ main (int argc, char **argv)
                 else break;                         // ...bail out
             }
             if (pollin[0].revents & POLLIN) { // New input, process it
-                if (fgets(input, sizeof(input), stdin) == NULL)
-                    break; // EOF received
-
+                *input = 0;
```
Comment (LemonBoy) [inline lemonbar.c]: `input[0] = '\0'`;

**Comment by LemonBoy** [comment]
Also, the commit messages should be reworded

* `fixed LemonBoy#107` means nothing, write something descriptive like `Consume all the lines before parsing`
* `Add WM_CLASS ATOM for compliance` compliant to what? `Add WM_CLASS atom` is fine here

Other than that the PR looks fine, /hopefully/ it fixes the problem @woho has (beside being a snark master, that's it)

**Comment by tryone144** [comment]
Reworded the commit messages and adopted coding style.

Additionally the WM\_CLASS is now properly set with "instance", "class".

|element|description|
|---|---|
|`class`|programm name, *bar* in this case|
|`instance`|programm instance, usually the executable name as passed by `argv[0]`|

**Comment by LemonBoy** [comment]
This PR is about fixing the multiline issue, open another one for the `WM_CLASS` thing.
Also, why there's a `.diff` in the changeset ?

**Comment by tryone144** [comment]
Sorry, I've forgot to check the last commit. I will split the PR into two: multiline issue and WM_CLASS.

**Comment by tryone144** [comment]
The WM_CLASS commits now have their own PR: https://github.com/LemonBoy/bar/pull/187

**Comment by alexd2580** [comment]
I object to merging this PR, since with these changes the same bug will happen as has been [observed by @LemonBoy](https://github.com/LemonBoy/bar/issues/107#issuecomment-80460169) in a slightly different context.

> Your patch seems to break my statusbar [1] for some reason
> 
> [1] http://ix.io/gTm

The reason for this breakage is that the statusbar script uses `echo -n` multiple times to generate one single line. Therefore the input buffer of lemonbar will be filled partially multiple times with small delays inbetween. 

With a blocking `fgets`, this problem mentioned above does not occur, because it waits for `EOS` or `\n` which is not sent until the complete line is terminated with an `echo` or with the statusbar script exiting and closing the pipe.

With a non-blocking `fgets` (proposed in this PR) the problem occurs: after each `echo -n` the input buffer is drained, thus reading, parsing and displaying a partial statusbar message.

While this does improve performance *and* fix the `{echo a; echo b} | lemonbar -p`-bug, as shown [here](https://github.com/krypt-n/bar/pull/32), it also introduces *race conditions* into `lemonbar`, when using it with any status generator that uses multiple statements (`echo`, `cout`) to generate one single `\n`-terminated line and is even a little bit more complex than `date | lemonbar`.

**Comment by tryone144** [comment]
This PR has been made redundant by the superior changes in https://github.com/LemonBoy/bar/commit/9e35bd6ccb9bd265322de960ff6276c4ec1af397. Please close.

**Pull request closed without merging**